drivers/overlay/overlay.go

```diff
@@ -1233,7 +1238,10 @@ func (d *Driver) recreateSymlinks() error {
 			if len(targetComponents) != 3 || targetComponents[0] != ".." || targetComponents[2] != "diff" {
 				errs = multierror.Append(errs, errors.Errorf("link target of %q looks weird: %q", link, target))
 				// force the link to be recreated on the next pass
-				os.Remove(filepath.Join(linksDir, link.Name()))
+				if err := os.Remove(filepath.Join(linksDir, link.Name())); err != nil {
```

Comment:
Should this error if the error is ENOEXIST?

Reply:
Good point — that shouldn’t be reported as an error, but it also shouldn’t be marked as having made progress.